src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -877,6 +877,13 @@ def build_vmss_resource(cmd, name, naming_prefix, location, tags, overprovision,
         }
         vmss_properties['virtualMachineProfile']['scheduledEventsProfile'] = scheduled_events_profile
 
+    if automatic_repairs_grace_period is not None:
+        automatic_repairs_policy = {
+            'enabled': 'true',
+            'gracePeriod': automatic_repairs_grace_period
```

Comment:
is grace period required property in rest api? if yes, it must be paired applied by user right? then need validation. is there default value? the 30 mins limitation is done at service side? If CLI hints user that constraint, would it be good to add constraint at CLI side? not sure what's the common practice there..

Reply:
From REST api perspective, it is not required. If it is omitted, service will give it a default value 30 to make sure the property has value.
Currently we only add --automatic-repairs-grace-period for `az vmss create` (no --enable-automatic-repairs). If user specify --automatic-repairs-grace-period, there means he/she want to enable automatic repairs. However this will cause that there will be no chance to trigger the service side default value logic.
I prefer our current design(not add --enable-automatic-repairs for `az vmss create`), because we have showed the value range in help message and it's better to make user know what value they are using explicitly. 
I'm not sure about this, so want to collect more suggestions here. 

For the question whether we should validate in CLI side, I prefer to not add. I think it's a duplicate work and service may change this logic later, if we add this validation, we also need to change according to service's changes later. Also want to listen your opinions.

Reply:
got it, thanks @arrownj .  Firstly, about parameters, `enable-automatic-repairs` and `automatic-repairs-grace-period` are different meaning, it's better to separate them to corresponding service api property instead of use implication that configuring grace period means enabling,
Secondly, for default value, yes it's more reliable to reply on service validation. If there's any description about value constraints at restapi spec, it will good if link that in help message to direct user to source of truth.

Reply:
I prefer not adding `--enable-automatic-repairs` in `vmss create`. `vmss update` has two arguments which is suitable. However, user cannot disable it since it's a create command. So it make senses to me that we just have one argument `--automatic-repairs-grace-period`.

Reply:
thanks @MyronFanQiu . we've offline discussed last Friday, will keep current behavior: use one parameter for setting grace period at creating, no enable parameter.